aptos-move/framework/aptos-experimental/sources/trading/market/market_types.move

```diff
@@ -74,18 +74,28 @@ module aptos_experimental::market_types {
         OrderStatus::ACKNOWLEDGED
     }
 
-    enum SettleTradeResult has drop {
+    enum CallbackResult<R: store + copy + drop> has copy, drop {
+        CONTINUE_MATCHING {
+            result: vector<R>
```

Comment:
why vector<R>, why is not vector inside R?

Reply:
This is vector because it allows merging of results across different matching steps. In case of `CONTINUE_MATCHING`, we want to merge the vectors and return the result to the caller.

Reply:
Individual `settle_trade` needs to return a vector as individual settlement can result in a bunch of different actions - which include cancelling a list of orders and placing a list of orders.